stl_to_obj/stl_to_tscn.py

```diff
@@ -0,0 +1,209 @@
+
+            # Call project operator to generate uv maps
+            try:
+                UV_MAPS[uv_map]()
+            except KeyError as e:
+                print(f"no uv map generated for {stl_file_extless} due to invalid uv_map")
+
+            # Toggle out of Edit Mode
+            bpy.ops.object.mode_set(mode='OBJECT')
+            # Deselect the object
+            obj.select_set(False)
+
+        # Export to obj
+        bpy.ops.wm.obj_export(filepath = f"{self.output_folder}/{stl_file_extless}.obj")
+        # remove the .mtl file
+        os.remove(f"{self.output_folder}/{stl_file_extless}.mtl")
+
+        # add to tscn file
+        self.nodes += f'[node name="{stl_file_extless}" type="MeshInstance" parent="."]\nmesh = ExtResource( {self.ext_resource_id} )' \
+                    + f'\nmaterial/0 = SubResource( {self.texture_index[texture]} )\n\n'
+        self.ext_resource += f'[ext_resource path="res://models/{self.output_folder}/{stl_file_extless}.obj" type="ArrayMesh" id={self.ext_resource_id}]\n'
+        self.ext_resource_id += 1
+
+    def add_collisions(self, stl_file_extless):
+        # generate fbx file from stl file to create collisions
+        fbx_file = f"{self.output_folder}/{stl_file_extless}.fbx"
+        scene = a3d.Scene.from_file(f"{self.input_folder}/{stl_file_extless}.stl")
+        scene.save(fbx_file)
+
+        # reset bpy
+        bpy.ops.wm.read_factory_settings(use_empty=True)
+        # import .fbx into bpy
+        bpy.ops.import_scene.fbx(filepath = fbx_file)
+
+        # rename all objects as -colonly
+        for obj_index in range(len(bpy.data.objects)):
+            # every re-name moves the object to the back of the list. 
+            bpy.data.objects[0].name = "Volume" + str(obj_index) + "-colonly"
+
+        # export to .gltf
+        bpy.ops.export_scene.gltf(filepath = f"{self.output_folder}/{stl_file_extless}.gltf")
+
+        # remove the fbx file
+        os.remove(fbx_file)
+
+        # add to tscn file  
+        self.nodes += f'[node name="{stl_file_extless}Col" parent="." instance=ExtResource( {self.ext_resource_id} )]\n\n'
+        self.ext_resource += f'[ext_resource path="res://models/{self.output_folder}/{stl_file_extless}.glb" type="PackedScene" id={self.ext_resource_id}]\n'
+        self.ext_resource_id += 1
+
+    def write_tscn_file(self):
+        with open(f"{self.output_folder}/{self.output_folder}.tscn", 'w') as tscn_file:
+            tscn_file.write(f"[gd_scene load_steps={self.ext_resource_id + self.sub_resource_id} format=2]\n\n{self.ext_resource}\n{self.sub_resource}{self.nodes}")
+
+
+def make_folder(folder):
+    try:
+        os.mkdir(f"./{folder}")
+    # delete the folder and remake it if it does exist
+    except OSError as error:
+        print(f"NOTICE: {folder} folder already exists, so it was deleted and remade")
+        shutil.rmtree(folder) # delete folder and contents
+        os.mkdir(f"./{folder}")
```

Comment:
Opportunity to implement better practices here for `try`, `except` blocks as mentioned above. Also note that raising an exception here will terminate the process. If that's not the intended functionality, try raising a warning instead.